src/DonorProfiles/Tabs/TabsRegister.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Give\DonorProfiles\Tabs;
+
+use Give\DonorProfiles\Tabs\Contracts\Tab;
+use http\Exception\InvalidArgumentException;
+
+class TabsRegister {
+	/**
+	 * FQCN of Tab classes
+	 *
+	 * @since 2.11.0
+	 *
+	 * @var string[]
+	 */
+	private $tabs = [];
+
+	/**
+	 * Returns all of the registered tabs
+	 *
+	 * @since 2.11.0
+	 *
+	 * @return string[]
+	 */
+	public function getTabs() {
+		return $this->tabs;
+	}
+
+	/**
+	 * Checks to see if a tab is registered with the given ID
+	 *
+	 * @since 2.11.0
+	 *
+	 * @param string $id
+	 *
+	 * @return bool
+	 */
+	public function hasTab( $id ) {
+		return isset( $this->tabs[ $id ] );
+	}
+
+	/**
+	 * Returns a tab with the given ID
+	 *
+	 * @since 2.11.0
+	 *
+	 * @param string $id
+	 *
+	 * @return string
+	 */
+	public function getTab( $id ) {
+		if ( ! isset( $this->tabs[ $id ] ) ) {
+			throw new \InvalidArgumentException( "No migration exists with the ID {$id}" );
+		}
+
+		return $this->migrations[ $id ];
+	}
+
+	/**
+	 * Returns all of the registered tab ids
+	 *
+	 * @since 2.11.0
+	 *
+	 * @return string[]
+	 */
+	public function getRegisteredIds() {
+		return array_keys( $this->tabs );
+	}
+
+	/**
+	 * Add a tab to the list of tabs
+	 *
+	 * @since 2.11.0
+	 *
+	 * @param string $tabClass FQCN of the Tab Class
+	 */
+	public function addTab( $tabClass ) {
+		if ( ! is_subclass_of( $tabClass, Tab::class ) ) {
```

Comment:
I would suggest type hinting the method parameter and let PHP enforce the sub-classing contract.

Reply:
Here, the addTab method is actually expecting a string, since what is being stored in the tabs property is just an array of fully qualified class names (as strings). This avoids constructing the classes themselves until they are needed. The manual check is required since we're really checking if the FQCN provided represents a class that does in fact extend the Tab class.

Reply:
Sounds good!